app/uk/gov/hmrc/fhregistrationfrontend/models/ModelEncryption.scala

```diff
@@ -51,25 +49,25 @@ object ModelEncryption {
   }
 
   def encryptSessionCache(
-    sessionCache: SummaryConfirmation
-  ): (String, Option[String], Option[WithdrawalReason], Option[DeregistrationReason]) =
+    sessionCache: SummaryConfirmationCache
+  )(implicit encryption: Encryption): (String, EncryptedValue, LocalDateTime, Instant) =
     (
       sessionCache.id,
-      sessionCache.summaryForPrintKey,
-      sessionCache.withdrawalReason,
-      sessionCache.deregistrationReason
+      encryption.crypto.encrypt(Json.toJson(sessionCache.fhSession).toString, sessionCache.id),
+      sessionCache.createdAt,
+      sessionCache.lastModified
     )
 
   def decryptSessionCache(
     id: String,
-    summaryForPrintKey: Option[String],
-    withdrawalReason: Option[WithdrawalReason],
-    deregistrationReason: Option[DeregistrationReason]
-  ): SummaryConfirmation =
-    SummaryConfirmation(
+    fhSession: EncryptedValue,
+    createdAt: LocalDateTime,
```

Comment:
Do we need `createdAt` and `lastModified` in here? (If they're not in the original case class, should they be in this one?

Reply:
I've removed them